tests/functional/partial_parsing/test_pp_vars.py

```diff
@@ -325,14 +323,20 @@ def test_profile_env_vars(self, project):
         os.environ["ENV_VAR_USER"] = "root"
         os.environ["ENV_VAR_PASS"] = "password"
 
-        results = run_dbt(["run"])
+        run_dbt(["run"])
         manifest = get_manifest(project.project_root)
         env_vars_checksum = manifest.state_check.profile_env_vars_hash.checksum
 
         # Change env_vars, the user doesn't exist, this should fail
         os.environ["ENV_VAR_USER"] = "fake_user"
-        (results, log_output) = run_dbt_and_capture(["run"], expect_pass=False)
+
+        # N.B. run_dbt_and_capture won't work here because FailedToConnectError ends the test entirely
+        with pytest.raises(FailedToConnectError):
+            run_dbt(["run"], expect_pass=False)
+
+        log_output = Path(project.project_root, "logs", "dbt.log").read_text()
```

Comment:
I ended up doing something similar in [my attempt to fix this](https://github.com/dbt-labs/dbt-core/pull/6908) but @gshank raises a really good point in [this comment](https://github.com/dbt-labs/dbt-core/pull/6908#discussion_r1100730065) - why is this test is now throwing an error in click-cli when it doesn't in main?